Sources/SwiftProtobuf/Message+BinaryAdditions.swift

```diff
@@ -34,18 +34,15 @@ public extension Message {
     let requiredSize = try serializedDataSize()
     var data = Data(count: requiredSize)
     try data.withUnsafeMutableBytes { (pointer: UnsafeMutablePointer<UInt8>) in
-      try serializeBinary(into: pointer)
+      var visitor = BinaryEncodingVisitor(forWritingInto: pointer)
+      try traverse(visitor: &visitor)
+      if requiredSize != visitor.encoder.distance(pointer: pointer) {
```

Comment:
I'm torn here, if this fires, it means there is a bug within the library, so should we just make this an assert instead of making a developer exposed error?